src/client/renderer/r_draw.c

```diff
@@ -22,40 +22,65 @@
 #include "r_local.h"
 #include "client.h"
 
-#define MAX_CHARS 16384  // per font
+#define MAX_CHARS MAX_GL_ARRAY_LENGTH  // per font
 // characters are batched per frame and drawn in one shot
 // accumulate coordinates and colors as vertex arrays
 typedef struct r_char_arrays_s {
-	GLshort verts[MAX_CHARS * 4 * 2];
+	GLfloat verts[MAX_CHARS * sizeof(vec3_t)];
 	uint32_t vert_index;
+	r_buffer_t vert_buffer;
 
-	GLfloat texcoords[MAX_CHARS * 4 * 2];
+	GLfloat texcoords[MAX_CHARS * sizeof(vec2_t)];
 	uint32_t texcoord_index;
+	r_buffer_t texcoord_buffer;
 
-	GLbyte colors[MAX_CHARS * 4 * 4];
+	GLfloat colors[MAX_CHARS * sizeof(vec4_t)];
```

Comment:
I can see why you're suggesting unsigned bytes for color attributes.. that's a lot of memory.